dlk/python/dlk/templates/include/quantizer.h

```diff
@@ -48,21 +48,16 @@ void func_QTZ_binary_channel_wise_mean_scaling(
   T_UINT in_channel)
 {
   unsigned num_elems_in_channel = in_height * in_width * in_depth;
-  T_FLOAT sum[in_channel];
-  T_FLOAT mean[in_channel];
-
-  for (unsigned i = 0; i < in_channel; i++) {
-    sum[i] = 0;
-  }
+  T_FLOAT sum;
 
   for(unsigned i = 0; i < in_channel; i++) {
+    sum = 0;
     for(unsigned j = 0; j < num_elems_in_channel; j++) {
-      sum[i] += std::abs(input[i * num_elems_in_channel + j]);
-    }
-    mean[i] = sum[i] / num_elems_in_channel;
-    for(unsigned j = 0; j < num_elems_in_channel; j++) {
-      unsigned in_index = i * num_elems_in_channel + j;
-      output[in_index] = (input[in_index] >= 0) ? mean[i] : -1 * mean[i];
+      sum += std::abs(input[i * num_elems_in_channel + j]);
+      for(unsigned j = 0; j < num_elems_in_channel; j++) {
+	unsigned in_index = i * num_elems_in_channel + j;
+	output[in_index] = (input[in_index] >= 0) ? sum / num_elems_in_channel : -1 * sum / num_elems_in_channel;
+      }
```

Comment:
are you sure you want to have two nested

~~~cpp

for(unsigned j = 0; j < num_elems_in_channel; j++) {
...
    for(unsigned j = 0; j < num_elems_in_channel; j++) {
        unsigned in_index = i * num_elems_in_channel + j;
        output[in_index] = ...
~~~
which will just overwrite each `output[in_index]` `num_elems_in_channel` times keeping only last value? :thinking: 

how about something like below instead?
```cpp
T_FLOAT sum = 0;
for(unsigned j = 0; j < num_elems_in_channel; j++) {
    sum += std::abs(input[i * num_elems_in_channel + j]);
}
T_FLOAT mean = sum / num_elements_in_channel;
for(unsigned j = 0; j < num_elems_in_channel; j++) {
    unsigned in_index = i * num_elems_in_channel + j;
    output[in_index] = (input[in_index] >= 0) ? mean : -1 * mean;
}
```
which is more straight forward transformation of initial code...

Reply:
You are right. This code is more simple. Thanks!